src/lib/utils.test.ts

```diff
@@ -224,19 +286,123 @@ describe('utils', () => {
       },
+      {
+        schema: {
+          name: 'TupleWithArray',
+          key: '0x4b80742de2bf393a64c70000<address>',
+          keyType: 'MappingWithGrouping',
+          valueType: '(bytes4,bytes8,bytes32[CompactBytesArray])',
+          valueContent: '(Bytes4,Number,Bytes32)',
+        },
+        valueType: '',
+        decodedValue: [
+          '0xdeadbeaf',
+          12,
+          [
+            '0x1234567812345678123456781234567812345678123456781234567812345678',
+            '0x2345678123456781234567812345678123456781234567812345678123456789',
+          ],
+        ] as string[] | Array<string | string[]>,
+        encodedValue:
+          '0xdeadbeaf000000000000000c0020123456781234567812345678123456781234567812345678123456781234567800202345678123456781234567812345678123456781234567812345678123456789',
+      },
+      {
+        schema: {
+          name: 'TupleWithArrayTruncated',
+          key: '0x4b80742de2bf393a64c70000<address>',
+          keyType: 'MappingWithGrouping',
+          valueType: '(bytes4,bytes8,bytes32[CompactBytesArray])',
+          valueContent: '(Bytes4,Number,Bytes32)',
+        },
+        valueType: '',
+        decodedValue: ['0xdeadbeaf', 12, null] as
+          | string[]
+          | Array<string | string[]>,
+        encodedValue: '0xdeadbeaf000000000000000c',
+      },
+      {
+        schema: {
+          name: 'TupleWithArrayPartial',
+          key: '0x4b80742de2bf393a64c70000<address>',
+          keyType: 'MappingWithGrouping',
+          valueType: '(bytes4,bytes8,bytes32[CompactBytesArray])',
+          valueContent: '(Bytes4,Number,Bytes32)',
+        },
+        valueType: '',
+        decodedValue: ['0xdeadbeaf', 12, null] as
+          | string[]
+          | Array<string | string[]>,
+        encodedValue: '0xdeadbeaf000000000000000c',
+        encodedError: '0xdeadbeaf000000000000000c0020', // This encoded value has the length item of the encoded array instead of null
+      },
+      {
+        schema: {
+          name: 'TupleWithBytes4Uint256',
+          key: '0x4b80742de2bf393a64c70000<address>',
+          keyType: 'Singleton',
+          valueType: '(bytes4,uint256)',
+          valueContent: '(Bytes4,Number)',
+        },
+        valueType: '',
+        decodedValue: ['0xc52d6008', 1] as string[] | Array<string | string[]>,
+        encodedValue:
+          '0xc52d60080000000000000000000000000000000000000000000000000000000000000001',
+      },
+      {
+        schema: {
+          name: 'TupleWithArrayPartialMore',
+          key: '0x4b80742de2bf393a64c70000<address>',
+          keyType: 'MappingWithGrouping',
+          valueType: '(bytes4,bytes8,bytes32[CompactBytesArray])',
+          valueContent: '(Bytes4,Number,Bytes32)',
+        },
+        valueType: '',
+        decodedValue: ['0xdeadbeaf', 12, null] as
```

Comment:
Couldn't we improve the typing for the provided inputs so we don't have to use `as`?

Reply:
I added some typedefs.